earthaccess/__init__.py

```diff
@@ -10,7 +10,6 @@
     get_requests_https_session,
     get_s3_credentials,
     get_s3fs_session,
-    granule_query,
```

Comment:
Can we just replace this entire import list (lines 5-17) with `from .api import *`, since that's mostly what's being defined in `__all__` anyway? That way we only need to maintain one extra copy of the list?

Reply:
@JessicaS11 I get the desire not to have duplication, and it *is* definitely tedious. However, `import *` obscures where objects in the namespace are actually coming from, leads to unexpected circular imports, and allows for namespace collisions that are hard to debug.

Becasue of that, I'd prefer to be explicit here as much as possible and live with the duplicaton. *If*, however, we did want to move to `import *` for building the API, we'll want to povide an `__all__` at the module level (e.g., `earthaccess.api.__all__`) so the imports only bring in the parts of the module we want. (`__all__` is for the level above it)

Reply:
@jhkennedy Thanks for the explanation - I wasn't sure if this was one of the "exceptions" when `import *` could more safely be used. We will stick to the manual approach then.